src/api/controllers/socialAuth.js

```diff
@@ -0,0 +1,110 @@
+/* eslint-disable no-else-return */
+import model from '../models';
+import generateToken from '../../helpers/tokens/generate.token';
+
+const { User } = model;
+
+/**
+ *
+ *
+ * @class socialLogin
+ */
+class socialLogin {
+  /**
+   * signup via facebook
+   * login  via facebook
+   *
+   * @static
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @memberof socialLogin
+   * @returns {object} - the response body
+   */
+  static async socialFacebookAuth(req, res) {
+    let findIfExist = await User.findOne({
+      where: {
+        uniqueId: req.user.id
+      }
+    });
+    if (!findIfExist) {
+      findIfExist = await User.create({
+        username: req.user.name.familyName,
+        profileImage: req.user.photos[0].value,
+        provider: req.user.provider,
+        uniqueId: req.user.id
+      });
+    }
+    const registerUser = { username: req.user.displayName, uniqueId: req.user.id };
+    const token = generateToken(registerUser, process.env.TOKEN_KEY);
+    return res.status(201).json({ user: { ...findIfExist.get(), token } });
+  }
+
+  /**
+   * signup user via Google
+   * login user via Google
+   *
+   * @static
+   * @param {object} req - request object
+   * @param {object} res - request response
+   * @memberof socialLogin
+   * @returns {object} - the response body
+   */
+  static async socialGoogleAuth(req, res) {
+    let findIfExist = await User.findOne({
+      where: {
+        uniqueId: req.user.id
+      }
+    });
+    if (!findIfExist) {
+      findIfExist = User.create({
+        username: req.user.displayName,
+        email: req.user.emails[0].value,
+        profileImage: req.user.photos[0].value,
+        provider: req.user.provider,
+        uniqueId: req.user.id
+      });
+      const registerUser = { username: req.user.displayName, email: req.user.emails[0].value };
+      const token = generateToken(registerUser, process.env.TOKEN_KEY);
+      return res.status(201).json({ user: { ...findIfExist.get(), token } });
+    } else {
+      const loginUser = { username: req.user.displayName, email: req.user.emails[0].value };
+      const token = generateToken(loginUser, process.env.TOKEN_KEY);
+      return res.status(201).json({ user: { ...findIfExist.get(), token } });
+    }
+  }
+
+  /**
+   * signup user via Twitter
+   * login user via Twitter
+   *
+   * @static
+   * @param {object} req - request object
+   * @param {object} res - request response
+   * @memberof socialLogin
+   * @returns {object} - the response body
+   */
+  static async socialtwitterAuth(req, res) {
```

Comment:
*Question:* do we currently have a uniqueId field in the user table? If we don't, you'll have to create a migration to add it.

*Feedback:* 
- The three social login methods have similarities. Can you abstract out the common parts into a separate module to keep the code DRY?

- `socialtwitterAuth` can simply be `twitterAuth` or `twitterLogin`.

- `findIfExist` isn't a good variable name. `existingUser` is better here.